rest_server/handlers.py

```diff
@@ -0,0 +1,206 @@
+
+else:
+    service_account_auth = handler.keycloak_role_auth
+
+
+# -----------------------------------------------------------------------------
+
+
+class BaseSkyDriverHandler(RestHandler):  # type: ignore  # pylint: disable=W0223
+    """BaseSkyDriverHandler is a RestHandler for all SkyDriver routes."""
+
+    def initialize(  # type: ignore  # pylint: disable=W0221
+        self,
+        mongodb_url: str,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a BaseSkyDriverHandler object."""
+        super().initialize(*args, **kwargs)
+        # pylint: disable=W0201
+        self.manifests = database.interface.ManifestClient(
+            AsyncIOMotorClient(mongodb_url)
+        )
+        self.results = database.interface.ResultClient(AsyncIOMotorClient(mongodb_url))
+
+    def prepare(self):
+        # TODO: put this into rest-tools
+        LOGGER.debug(
+            f"{self.request.path} {self.request.method}("
+            f"{self.path_args}, {self.path_kwargs}) [{self.__class__.__name__}]"
+        )
+        super().prepare()
+
+
+# -----------------------------------------------------------------------------
+
+
+class MainHandler(BaseSkyDriverHandler):  # pylint: disable=W0223
+    """MainHandler is a BaseSkyDriverHandler that handles the root route."""
+
+    ROUTE = r"/$"
+
+    @service_account_auth(roles=[USER_ACCT])  # type: ignore
+    async def get(self) -> None:
+        """Handle GET."""
+        self.write({})
+
+
+# -----------------------------------------------------------------------------
+
+
+class EventMappingHandler(BaseSkyDriverHandler):  # pylint: disable=W0223
+    """Handles mapping an event to scan(s)."""
+
+    ROUTE = r"/event/(?P<event_id>\w+)$"
+
+    @service_account_auth(roles=[USER_ACCT])  # type: ignore
+    async def get(self, event_id: str) -> None:
+        """Get matching scan id(s) for the given event id."""
+        incl_del = self.get_argument("include_deleted", default=False, type=bool)
+
+        scan_ids = [s async for s in self.manifests.get_scan_ids(event_id, incl_del)]
+
+        self.write({"event_id": event_id, "scan_ids": scan_ids})
+
+    #
+    # NOTE - 'EventMappingHandler' needs to stay user-read-only b/c
+    #         it's indirectly updated by the launching of a new scan
+    #
+
+
+# -----------------------------------------------------------------------------
+
+
+class ScanLauncherHandler(BaseSkyDriverHandler):  # pylint: disable=W0223
+    """Handles starting new scans."""
+
+    ROUTE = r"/scan$"
+
+    @service_account_auth(roles=[USER_ACCT])  # type: ignore
+    async def post(self) -> None:
+        """Start a new scan."""
+
+        def valid_event_id(val: Any) -> str:
+            val = str(val)
+            if any(x.isspace() for x in val) or not any(x.isalnum() for x in val):
+                raise TypeError(
+                    "event_id must contain alphanumeric characters and no spaces"
+                )
+            return val
```

Comment:
Couldn't this whole function be replaced by `str(val).isalnum()`?  That already checks for spaces and can operate on a whole string.

Reply:
I replaced the error message with "must contain no spaces and at least one alphanumeric character"

Reply:
I figure event ids like `abc.2` and `foobar[baz]` are okay

Reply:
Maybe.  The check is just really vague right now.  Would be nice to get this tightened up somehow.

Reply:
I'm not sure what we want to allow in an id. For the actual logic, preferably this would be a `re.match`

Reply:
Is this supposed to be an i3 event id, or something we make up?